filter/format_time.go

```diff
@@ -0,0 +1,218 @@
+	rfc3339nano = "RFC3339Nano"
+	unix        = "unix"
+	unixms      = "unixms"
+	unixns      = "unixns"
+)
+
+var FormatTimeDesc = baker.FilterDesc{
+	Name:   "FormatTime",
+	New:    NewFormatTime,
+	Config: &FormatTimeConfig{},
+	Help:   formatTimeHelp,
+}
+
+type FormatTimeConfig struct {
+	SrcField  string `help:"Field name of the input time" required:"true"`
+	DstField  string `help:"Field name of the output time" required:"true"`
+	SrcFormat string `help:"Format of the input time" required:"false" default:"UnixDate"`
+	DstFormat string `help:"Format of the output time" required:"false" default:"unixms"`
+}
+
+func (cfg *FormatTimeConfig) fillDefaults() {
+	if cfg.SrcFormat == "" {
+		cfg.SrcFormat = unixdate
+	}
+	if cfg.DstFormat == "" {
+		cfg.DstFormat = unixms
+	}
+}
+
+type FormatTime struct {
+	src    baker.FieldIndex
+	dst    baker.FieldIndex
+	parse  func(t []byte) (time.Time, error)
+	format func(t time.Time) []byte
+
+	// Shared state
+	numProcessedLines int64
+}
+
+func NewFormatTime(cfg baker.FilterParams) (baker.Filter, error) {
+	dcfg := cfg.DecodedConfig.(*FormatTimeConfig)
+	dcfg.fillDefaults()
+
+	f := &FormatTime{}
+
+	idx, ok := cfg.FieldByName(dcfg.SrcField)
+	if !ok {
+		return nil, fmt.Errorf("unknown field %q", dcfg.SrcField)
+	}
+	f.src = idx
+
+	idx, ok = cfg.FieldByName(dcfg.DstField)
+	if !ok {
+		return nil, fmt.Errorf("unknown field %q", dcfg.DstField)
+	}
+	f.dst = idx
+
+	f.parse = genParseFun(dcfg.SrcFormat)
+	f.format = genFormatFun(dcfg.DstFormat)
+
+	return f, nil
+}
+
+func (f *FormatTime) Stats() baker.FilterStats {
+	return baker.FilterStats{
+		NumProcessedLines: atomic.LoadInt64(&f.numProcessedLines),
+	}
+}
+
+func (f *FormatTime) Process(l baker.Record, next func(baker.Record)) {
+	atomic.AddInt64(&f.numProcessedLines, 1)
+
+	t, err := f.parse(l.Get(f.src))
+	if err != nil {
+		log.Errorf("can't parse time: %v", err)
+		l.Set(f.dst, nil)
+	} else {
+		l.Set(f.dst, f.format(t))
+	}
+
+	next(l)
+}
+
+func formatToLayout(format string) string {
+	switch format {
+	case ansic:
+		return time.ANSIC
+	case unixdate:
+		return time.UnixDate
+	case rubydate:
+		return time.RubyDate
+	case rfc822:
+		return time.RFC822
+	case rfc822z:
+		return time.RFC822Z
+	case rfc850:
+		return time.RFC850
+	case rfc1123:
+		return time.RFC1123
+	case rfc1123z:
+		return time.RFC1123Z
+	case rfc3339:
+		return time.RFC3339
+	case rfc3339nano:
+		return time.RFC3339Nano
+	default:
+		return format
+	}
+}
+
+func genParseFun(format string) func(b []byte) (time.Time, error) {
+	switch format {
+	case unix:
+		return func(b []byte) (time.Time, error) {
+			sec, err := strconv.ParseInt(string(b), 10, 64)
+			if err != nil {
+				return time.Time{}, err
+			}
+			return time.Unix(sec, 0), nil
+		}
+	case unixms:
+		return func(b []byte) (time.Time, error) {
+			msec, err := strconv.ParseInt(string(b), 10, 64)
+			if err != nil {
+				return time.Time{}, err
+			}
+			return time.Unix(0, msec*1_000_000), nil
```

Comment:
there are constants in the `time` package for that:
```suggestion
			return time.Unix(0, msec*int64(time.Millisecond)), nil
```